### ISSUE TITLE
_token_to_eth_swap_input checks price instead of balance

### ISSUE THREAD

**Issue by CooleRnax** [shanefontaine/uniswap-python#25]
I have discovered a bug that _token_to_eth_swap_input checks price instead of balance and throws an error.

```
    tx = self.make_trade(address_in, address_out, amount_in)
  File "../../../libs/crypto_api\crypto_api\uniswap.py", line 66, in approved
    return method(self, *args)
  File "../../../libs/crypto_api\crypto_api\uniswap.py", line 200, in make_trade
    return self._token_to_eth_swap_input(input_token, qty, recipient)
  File "C:\ProgramData\Anaconda3\lib\site-packages\uniswap\uniswap.py", line 505, in _token_to_eth_swap_input
    raise InsufficientBalance(input_balance, cost)
uniswap.uniswap.InsufficientBalance: Insufficient balance. Had 115549539029086413, needed 137927441360803636
```

This is how it can be fixed:
```
    def _token_to_eth_swap_input(
        self, input_token: AddressLike,
        qty: int, recipient: Optional[AddressLike]
    ) -> HexBytes:
        """Convert tokens to ETH given an input amount."""
        # Balance check
        input_balance = self.get_token_balance(input_token)
        if qty > input_balance:
            raise InsufficientBalance(input_balance, qty)

        if self.version == 1:
            token_funcs = self.exchange_contract(input_token).functions
            func_params: List[Any] = [qty, 1, self._deadline()]
            if not recipient:
                function = token_funcs.tokenToEthSwapInput(*func_params)
            else:
                func_params.append(recipient)
                function = token_funcs.tokenToEthTransferInput(*func_params)
            return self._build_and_send_tx(function)
        else:
            if recipient is None:
                recipient = self.address
            return self._build_and_send_tx(
                self.router.functions.swapExactTokensForETH(
                    qty,
                    int((1 - self.max_slippage) * qty),
                    [input_token, self.get_weth_address()],
                    recipient,
                    self._deadline(),
                ),
            )
```

**Comment by CooleRnax** [comment]
Actually output amount needs to be calculated, by solution does not fully solve the problem

**Comment by ErikBjare** [comment]
Should be fixed in #30, feel free to take a look and try it out @CooleRnax
